packages/react-storage/src/components/StorageBrowser/Views/LocationDetailView/Controls/DataTable.tsx

```diff
@@ -0,0 +1,275 @@
+import React from 'react';
+
+import { humanFileSize } from '@aws-amplify/ui';
+
+import {
+  DataTable,
+  TABLE_DATA_BUTTON_CLASS,
+  TABLE_DATA_CLASS_NAME,
+  TABLE_DATA_ICON_CLASS,
+  TABLE_DATA_TEXT_CLASS_NAME,
+  TABLE_HEADER_BUTTON_CLASS_NAME,
+  TABLE_HEADER_CLASS_NAME,
+} from '../../../components/DataTable';
+import { useControl } from '../../../context/controls';
+import { useAction } from '../../../context/actions';
+import { LocationItem } from '../../../context/types';
+import {
+  compareDates,
+  compareNumbers,
+  compareStrings,
+} from '../../../context/controls/Table';
+import {
+  ButtonElement,
+  IconElement,
+  SpanElement,
+} from '../../../context/elements';
+import { DownloadControl } from '../../Controls';
+import { getFileExtension } from '../../utils';
+
+export type SortDirection = 'ascending' | 'descending' | 'none';
+
+export type SortState = {
+  selection: string;
+  direction: SortDirection;
+};
+
+type Column<T> = {
+  key: keyof T;
+  header: string;
+  compareFn?: (a: T, b: T) => number;
+};
+
+type DataTableLocationItems = {
+  key: string;
+  fileExt: string;
+  lastModified: Date | undefined;
+  size: number | undefined;
+  type: string;
+};
```

Comment:
Few things:
- these should be declared as interfaces, not types
- why are we reintroducing a `Column` interface? This isn't needed for the `LocationsView` right?

Reply:
added it to help with managing the `key` and `header` properties since I wanted to add a new column, `fileExt` that would contain the file extensions, but display as as a `type` column.

I think when I tried to set it up similar to how `LocationsView` was set up, I was running into a lot of typescript errors related to the sorting, but I can try it out again

Reply:
still working on this